frontend/src/tree/TreeItemContextMenu.tsx

```diff
@@ -44,60 +73,114 @@ export const TreeItemContextMenu = ({
     }
   };
 
+  const [deleteTreeItem, { loading: deleteTreeItemLoading, data: deleteTreeItemData, error: deleteTreeItemError }] =
+    useMutation<GQLDeleteTreeItemData, GQLDeleteTreeItemVariables>(deleteTreeItemMutation);
+
+  const deleteItem = () => {
+    const input: GQLDeleteTreeItemInput = {
+      id: uuid(),
+      editingContextId,
+      representationId: treeId,
+      treeItemId: item.id,
+    };
+    deleteTreeItem({ variables: { input } });
+    onClose;
+  };
+
+  const handleError = (loading, data, error) => {
```

Comment:
The build is right, the whole content of the method should be inside of the useEffect, don't separate the code in another method.